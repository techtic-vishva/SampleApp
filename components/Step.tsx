import React from 'react'
import { View } from 'react-native'

export default function Step(active: number, total: number = ONBOARDING_STEPS, style: {}) {
	const step = active - 1 // 0 based indexing
	return (
		<View style={[{ flexDirection: 'row' }, style]}>
			{Array.from(Array(total)).map((_, i) => {
				const active = step === i
				return (
					<View
						key={i}
						style={{
							width: active ? 60 : 5,
							borderRadius: 5,
							backgroundColor: 'white',
							height: active ? 3 : 5,
							marginHorizontal: 5,
							opacity: active ? undefined : 0.5,
							marginTop: active ? 1 : 0
						}}
					/>
				)
			})}
		</View>
	)
}

export const ONBOARDING_STEPS = 4
