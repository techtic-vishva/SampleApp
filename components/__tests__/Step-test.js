import * as React from 'react'
import renderer from 'react-test-renderer'

import Step from '../Step'

it('render when passing the params', () => {
	const tree = renderer.create(<Step active={1} total={4} style={{}} />).toJSON()
	expect(tree).toMatchSnapshot()
})
