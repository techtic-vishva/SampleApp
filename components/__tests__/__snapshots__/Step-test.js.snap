// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when passing the params 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 5,
        "height": 5,
        "marginHorizontal": 5,
        "marginTop": 0,
        "opacity": 0.5,
        "width": 5,
      }
    }
  />
</View>
`;
